### ISSUE TITLE
ElasticSearch 6 support

### ISSUE THREAD

**Issue by pschulten** [kubernetes/heapster#1909]
We're logging to ES 6 and the heapster export fails with:

`W1211 13:30:19.368537       1 driver.go:75] Failed to export data to ElasticSearch sink: elastic: Error 400 (Bad Request): Failed to parse mapping [general]: No handler for type [string] declared on field [cluster_name] [type=mapper_parsing_exception]`

**Comment by tstoermer** [comment]
ES 6 does no longer support multiple types in mappings per index. The metrics must be merged to one type or written to different indices.
https://www.elastic.co/guide/en/elasticsearch/reference/6.0/breaking-changes-6.0.html

There is also a new version of the used ES client 
https://github.com/olivere/elastic

**Comment by DirectXMan12** [comment]
cc @AlmogBaku @huangyuqi @andyxning

**Comment by AlmogBaku** [comment]
I'll try to find the time to make it supports ES6. If someone can pick it up please comment

**Comment by Adriien-M** [comment]
We are looking for this update, what is the status of this issue?

**Comment by Freyert** [comment]
@AlmogBaku I'm postulating that [these tests](https://github.com/kubernetes/heapster/blob/c079c2ab484cc709f336e3cf69d6f32dc180869c/metrics/sinks/elasticsearch/driver_test.go#L192-L197) represent the original schema?

@tstoermer is [this the documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/removal-of-types.html) you were referencing?

**Comment by tstoermer** [comment]
@Freyert 
Right, just wanted to point the official breaking changes - linking to the details you mentioned.

**Comment by ihardzeenka** [comment]
Hi guys. About schema for ES 6. 
Now you have single index with mapping types like cpu, memory, events and etc.
According to ES documentation there is 2 options here (https://www.elastic.co/guide/en/elasticsearch/reference/6.x/removal-of-types.html):
1) Simply split single index into multiple indexes like: heapster-cpu , heapster-memory and etc.
2) Merge types into single type (I don't really like that option)
I don't know go language but looking into your code I believe it is not a big change to simply add existing type name into index name.

**Comment by MnrGreg** [comment]
Would be great to get something going here!

**Comment by jimk-osu** [comment]
+1

**Comment by jhorwit2** [comment]
@DirectXMan12 what's the deprecation plan for eventer since that's not included functionality in the metrics-server?

**Comment by smelchior** [comment]
we do currently have the same issue, maybe https://www.elastic.co/guide/en/beats/metricbeat/6.x/metricbeat-metricset-kubernetes-event.html can be used as an alternative?

**Comment by DirectXMan12** [comment]
yeah, there's a few different solutions, depending on what your backend is.  Someone could also try to adapt https://github.com/GoogleCloudPlatform/k8s-stackdriver/tree/master/event-exporter, or fork eventer.
